Update book appointment endpoint params

ollyswanson opened the pull request.

ollyswanson committed:
Address uprn and SoR code domain types

There are repeated assertions about the validity of SoR codes and
address uprns (`locationId`) in the code, therefore we create types to
protect against this in the code and remove the need for tests at every
callsite.

ollyswanson committed:
Remove DRS code

ollyswanson committed:
Replace usages of primitives with proper types

This should remove the need for many of the tests that are in the
codebase, and provide better typing for method parameters and
invocations.

ollyswanson committed:
Relocate existing Dtos to Dtos/Mcm and use records

Records better capture the fact that these are immutable DTOs

ollyswanson committed:
Add `ArgumentOutOfRange` check to `AppointmentSlot`

This check is currently happening in one of the use cases but could live
on `AppointmentSlot` to avoid the need for testing in it in many places.

ollyswanson committed:
Update BookAppointment endpoint to take a new body

The MCM scheduling system requires a lot more information for scheduling
jobs/appointments than the scheduling system used in "other" councils,
therefore we must update the endpoint to take extra information which
can then be passed to MCM.

ollyswanson committed:
Update domain and usecases to use proper types

Offloading validation to the domain types allows us to remove the
usecase tests that are simply checking that the data is valid.

ollyswanson committed:
Temporarily return id without doing work

ollyswanson committed:
Remove `Name` field from `Contact`

Although MCM wants this data, we aren't collecting it and therefore
can't send it...

jamiejordanmt commented on HousingRepairsSchedulingApi/Domain/Contact.cs:
```diff
@@ -0,0 +1,19 @@
+/// </summary>
+public class Contact
+{
+    public Contact(string name, string phoneNumber, string notificationMobileNumber, string notificationEmail)
```
We're not collecting a name

jamiejordanmt commented on HousingRepairsSchedulingApi/Gateways/IAppointmentsGateway.cs:
```diff
@@ -1,15 +1,18 @@
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using Domain;
+using JetBrains.Annotations;
```
Is there a better way of doing this?

jamiejordanmt commented on HousingRepairsSchedulingApi/HousingRepairsSchedulingApi.csproj:
```diff
@@ -25,4 +25,8 @@
   </ItemGroup>
 
+  <ItemGroup>
+    <_ContentIncludedByDefault Remove="Services\Drs\ServiceReference\dotnet-svcutil.params.json" />
```
Where is this being used?

ollyswanson committed:
Replace Jetbrains imports with C# ones

ollyswanson committed:
Remove name field from request

We aren't collecting the name from the tenant, therefore we can't send
it as part of the request.

ollyswanson committed:
Remove changes inserted by IDE
